twill11/hometask-6/lib/parser.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+require 'http'
+require 'json'
+require_relative 'photo'
+
+# Parsing data from API
+class Parser
+  URL = 'https://www.flickr.com/services/rest/'
+  API_KEY = '5d54f2b42d120b834e3944afe7281069'
+  SEARCH_METHOD = 'flickr.photos.search'
+  GET_METHOD = 'flickr.photos.getInfo'
+  FORMAT = 'format=json&nojsoncallback=1'
+
+  def data_scrap(tag, count)
+    result = []
+    photos_search(tag, count).each do |photo|
+      id, title = photo.values_at('id', 'title')
+      owner, urls = get_info(id).values_at('owner', 'urls')
+      result << photo_dataset(Photo.new(id, title, owner['username'], urls['url'].first['_content']))
+    end
```

Comment:
Add to result Photo instances only.

```suggestion
      owner, urls = get_info(id).values_at('owner', 'urls')
      result << Photo.new(id, title, owner['username'], urls['url'].first['_content'])
    end
```